### ISSUE TITLE
Check for parentheses when within parameter list

### ISSUE THREAD

**Issue by agrimm** [bbatsov/rubocop#2426]
The following code doesn't create an offense:

    # Does fooing
    class Foo
      def needs_parentheses
        foo(bar 1, 2)
      end
    end

Should it do so? I think it's too easy to mistake it as saying `foo(bar, 1, 2)`.

**Comment by alexdowad** [comment]
I think whatever rules as regards parenthesizing method calls are applied elsewhere, should also be applied here in just the same way. It doesn't deserve a special case.

**Comment by bbatsov** [comment]
There are very few rules that are enforced automatically about parens. Mostly because it's really hard to determine if something is a DSL method (which generally are invoked without parens).  

I think @agrimm is right that the code he presents in the example is a bit unclear.

**Comment by jonas054** [comment]
I think a new cop for this special case would be a good idea. A method call without parentheses that's an argument in another method call is a weird thing.

**Comment by alexdowad** [comment]
I have a cop for this, waiting for my outstanding PR to be merged before I open another one.

**Comment by alexdowad** [comment]
The new cop flags this:

```ruby
 expect(slice.any? { |l| l.include? cop.cop_name }).to be_truthy
```

...because of the `#include?` call. Do you like this? Or should this be exempt, perhaps because it is inside a block?

**Comment by BenMorganIO** [comment]
Hmmm I'm thinking this was a bit too hastey. I got some code that looks like:

```ruby
foo(bar 'hello')
```

I consider that quite OK since you can tell there's an inside function. I'm thinking that this cop should only be on if the arity is greater than 1.

**Comment by alexdowad** [comment]
> I consider that quite OK since you can tell there's an inside function. I'm thinking that this cop should only be on if the arity is greater than 1.

Rather than adding more and more rules about when parens are required, I suggest we keep this cop simple. People who don't like the simple rule can disable the cop.
